acq4/devices/NiDAQ/resample_test.py

```diff
@@ -1,30 +1,33 @@
 # -*- coding: utf-8 -*-
 from __future__ import print_function
-import scipy.signal
-from numpy import *
-import numpy
-import time, sys
+
+import sys
+import time
+
+import numpy as np
+
+from ...pyqtgraph import PlotWindow
 
 sys.path.append('../../util')
 sys.path.append('../..')
 from .nidaq import NiDAQ
-from acq4.pyqtgraph.graphicsWindows import *
-from acq4.pyqtgraph.functions import mkPen
+from acq4.pyqtgraph.functions import mkPen, mkColor
+
 pw = PlotWindow()
 
 
 time.clock()
 sr = 100000
 dur = 2.0
-data = zeros(int(sr*dur))
+data = np.zeros(int(sr*dur))
 dlen = len(data)
-xVals = linspace(0, dur, dlen)
+xVals = np.linspace(0, dur, dlen)
 
-data += random.normal(size=dlen) + 20.
-data[dlen*0.102:dlen*0.3] += 20
-data[dlen*0.3:dlen*0.5] += 30
-data[dlen*0.4]+= 1000
-data += sin(xVals*40677*2.0*pi)*4.
+data += np.random.normal(size=dlen) + 20.
+data[round(dlen*0.102):round(dlen*0.3)] += 20
```

Comment:
Shouldn't these be `int` instead of `round` ?

Reply:
Good question! I didn't actually look up what the previous behavior would have been, but the math worked out that all of these `dlen * blah` expressions were supposed to be whole numbers, so if `int` and `round` had different answers in these cases due to floating point noise, `round` would almost certainly be the intended result. It may well be the case that the previous behavior was actually `int`, in which case this change probably fixes an unnoticed bug.

Reply:
Not just FP noise; `round(x)` and `int(x)` give different answers whenever `x % 1 > 0.5`. This particular example should work either way, but the previous numpy behavior (now deprecated) was to coerce float indices using `int`. Not a bug; it was the expected behavior.

Reply:
Mm, yes, I did check first that none of the rounded numbers were supposed to have decimal parts.